### PR TITLE
Move checking git repo size change to a separate test

### PR DESCRIPTION
**Pull request by iarspider**
I will add/modify Jenkins job once this PR is approved.

### CODE REVIEW

File: run-pr-code-checks [hunk 1]
```diff
@@ -57,11 +57,6 @@ DSIZE=0
 source $CMS_BOT_DIR/jenkins-artifacts
 pushd $CMSSW_BASE/src
   git checkout -b codechecks-${PULL_REQUEST} $CMSSW_VERSION
```
Comment (smuzaffar) [inline run-pr-code-checks:59]: @iarspider , I think this command is also not needed

File: run-pr-git-repo-size [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash -ex
```
Comment (smuzaffar) [inline run-pr-git-repo-size:1]: this check should be part of https://github.com/cms-sw/cms-bot/blob/master/pr_testing/test_multiple_prs.sh#L259

File: pr_testing/test_multiple_prs.sh [hunk 3]
```diff
@@ -786,6 +787,30 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
...
 
+  if $PRODUCTION_RELEASE ; then
+    pushd ..
+      mv src src.tmp
+      mkdir src && cd src
+      DSIZE=0
+      THRDS=""
+      NUM_PROC=$(nproc)
+      if [ $NUM_PROC = "0" ] ; then NUM_PROC=1; fi
+      if $USE_IB_TAG ; then git cms-init --upstream-only $CMSSW_IB ; else git cms-init --upstream-only ; fi
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:798]: @iarspider , replace this with `git cms-init --upstream-only && git checkout -b codechecks $CMSSW_IB` ( this is what run-pr-code-checks script was doing)

File: pr_testing/test_multiple_prs.sh [hunk 4]
```diff
@@ -786,6 +787,30 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
     exit 0
   fi
 
+  if $PRODUCTION_RELEASE ; then
+    pushd ..
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:791]: change this to `pushd $CMSSW_BASE`

File: pr_testing/test_multiple_prs.sh [hunk 5]
```diff
@@ -786,6 +787,30 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
...
+      mv src src.tmp
+      mkdir src && cd src
+      DSIZE=0
+      THRDS=""
+      NUM_PROC=$(nproc)
+      if [ $NUM_PROC = "0" ] ; then NUM_PROC=1; fi
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:797]: use `$NCPU` which is already set

File: pr_testing/test_multiple_prs.sh [hunk 6]
```diff
@@ -786,6 +787,30 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
...
+      if $USE_IB_TAG ; then git cms-init --upstream-only $CMSSW_IB ; else git cms-init --upstream-only ; fi
+      git repack -h 2>&1 | grep '\-\-threads' && THRDS="--threads ${NUM_PROC}" || true
+      git repack -a -d ${THRDS}
+      git repack -a -d ${THRDS}
+      OSIZE=$(du -sk .git/objects/pack | sed 's|\s.*||')
+      PR_NR=$(echo ${PULL_REQUEST} | sed 's/.*#//' )
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:803]: @iarspider , are we sure `${PULL_REQUEST}` is a cmssw PR? This whole section should be run for ` $PRODUCTION_RELEASE` and ` ${PULL_REQUEST}` being cms-sw/cmssw PR

File: pr_testing/test_multiple_prs.sh [hunk 7]
```diff
@@ -752,7 +752,8 @@ echo '{}' > $RECENT_COMMITS_FILE
 touch $WORKSPACE/changed-files
 if [ ! -d $WORKSPACE/cms-prs ]  ; then git clone --depth 1 git@github.com:cms-sw/cms-prs $WORKSPACE/cms-prs ; fi
 if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
-  if $USE_IB_TAG ; then git cms-init --upstream-only $CMSSW_IB ; fi
+  if $USE_IB_TAG ; then git cms-init --upstream-only $CMSSW_IB ; else git cms-init --upstream-only ; fi
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:755]: @iarspider , why was this needed?

File: pr_testing/test_multiple_prs.sh [hunk 8]
```diff
@@ -786,6 +787,28 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
...
+      mkdir src && cd src
+      DSIZE=0
+      THRDS=""
+      git cms-init --upstream-only && git checkout -b codechecks $CMSSW_IB
+      git repack -h 2>&1 | grep '\-\-threads' && THRDS="--threads ${NCPU}" || true
+      git repack -a -d ${THRDS}
+      git repack -a -d ${THRDS}
+      OSIZE=$(du -sk .git/objects/pack | sed 's|\s.*||')
+      PR_NR=$(echo ${PULL_REQUEST} | sed 's/.*#//' )
+      git cms-merge-topic --debug --ssh -u ${CMSSW_ORG}:${PR_NR}
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:802]: @iarspider , instead of `PR_NR`, please use `PR_NUMBER` which is already set

**Comment by smuzaffar** [comment]
please test

File: pr_testing/test_multiple_prs.sh [hunk 9]
```diff
@@ -786,6 +787,27 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
     exit 0
   fi
 
+  if [[ $PRODUCTION_RELEASE -a ${PULL_REQUEST} == "*/cmssw#*" ]]; then
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:790]: @iarspider , looks like this is not working

**Comment by smuzaffar** [comment]
please test

File: pr_testing/test_multiple_prs.sh [hunk 10]
```diff
@@ -786,6 +787,27 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
     exit 0
   fi
 
+  if [[ $PRODUCTION_RELEASE == "true" && ${PULL_REQUEST} == "*/cmssw#*" ]]; then
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:790]: @iarspider , looks like this check is still not working. Can you please test locally a simple script to test it before updating it here?

File: pr_testing/test_multiple_prs.sh [hunk 11]
```diff
@@ -786,6 +787,27 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
...
+      git cms-init --upstream-only && git checkout -b codechecks $CMSSW_IB
+      git repack -h 2>&1 | grep '\-\-threads' && THRDS="--threads ${NCPU}" || true
+      git repack -a -d ${THRDS}
+      git repack -a -d ${THRDS}
+      OSIZE=$(du -sk .git/objects/pack | sed 's|\s.*||')
+      git cms-merge-topic --debug --ssh -u ${CMSSW_ORG}:${PR_NUMBER}
+      git repack -d ${THRDS}
+      NSIZE=$(du -sk .git/objects/pack | sed 's|\s.*||')
+      let DSIZE=${NSIZE}-${OSIZE} || DSIZE=0
+      if [ $DSIZE -gt 0 ]; then echo -e "\n* - This PR adds an extra ${DSIZE}KB to repository" > ${RESULTS_DIR}/16-git-repo-size-report.res; fi
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:805]: @iarspider , please change this to 
```
if [ $DSIZE -gt 0 ]; then echo "**Size**: This PR adds an extra ${DSIZE}KB to repository" > ${RESULTS_DIR}/09-git-repo-size-report.res; fi
```

this will return the summary as https://cmssdt.cern.ch/SDT/cgi-bin/get_pr_results/jenkins-artifacts/pull-request-integration/PR-7b41ad/40198/pr-result

#############################

**Failed Tests**: RelVals 
**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7b41ad/40198/summary.html
**COMMIT**: 015d3ae04237944ad58e338ff7e259160cf5a536
**CMSSW**: CMSSW_14_1_X_2024-07-02-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45320/40198/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


## RelVals

- [29634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7b41ad/40198/runTheMatrix-results/29634.911_TTbar_14TeV+2026D110_DD4hep)```A fatal system signal has occurred: segmentation violation```

File: pr_testing/test_multiple_prs.sh [hunk 12]
```diff
@@ -786,6 +787,27 @@ if ! $CMSDIST_ONLY ; then # If a CMSSW specific PR was specified #
     exit 0
   fi
 
+  if [[ $PRODUCTION_RELEASE == "true" && ${PULL_REQUEST} == *"/cmssw#"* ]]; then
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:790]: Do we need quote in `*"/cmssw#"*`  ?

Comment (iarspider) [reply pr_testing/test_multiple_prs.sh:790]: It works with and without quotes.

Comment (smuzaffar) [reply pr_testing/test_multiple_prs.sh:790]: `*"/cmssw#"*` looks bit odd unless there is any recommendation to use quotes I would suggest to remove these

**Comment by smuzaffar** [comment]
https://github.com/cms-sw/cmssw/pull/45320#issuecomment-2208910391 looks good now
